File: python/0976.py
# Link: https://leetcode.com/problems/largest-perimeter-triangle/description/
# Runtime: 7 ms
# Memory: 18.69 MB

class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        copy_nums = nums[:]
        copy_nums.sort(reverse=True)
        for i in range(len(copy_nums) - 2):
            if copy_nums[i] < copy_nums[i + 1] + copy_nums[i + 2]:
                return copy_nums[i] + copy_nums[i + 1] + copy_nums[i + 2]

        return 0

File: c++/0976.cpp
// Link: https://leetcode.com/problems/largest-perimeter-triangle/
// Runtime: 18 ms
// Memory: 29.91 MB

class Solution {
public:
  int largestPerimeter(const std::vector<int> &nums) {
    auto copyNums = nums;
    std::sort(copyNums.rbegin(), copyNums.rend());
    for (int i = 0; i < static_cast<int>(copyNums.size() - 2); i++) {
      if (copyNums[i] < copyNums[i + 1] + copyNums[i + 2]) {
        return copyNums[i] + copyNums[i + 1] + copyNums[i + 2];
      }
    }

    return 0;
  }
};
